src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,668 @@ public ByteArraySet createSmallSet() {
+    }
+
+    @Test
+    public void testUnmergedSimilarRemove() {
+        testSimilarRemove(false);
+    }
+
+    public void testToArray(int size) {
+        System.out.println("Testing toArray method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        if (size > 0) {
+            assertFalse(set.isEmpty());
+        } else {
+            assertTrue(set.isEmpty());
+        }
+
+        List<ByteArray> alreadySeenVals = new ArrayList<ByteArray>();
+        int seenCount = 0;
+        for(ByteArray val: set.toArray()) {
+            assertFalse(alreadySeenVals.contains(val));
+            alreadySeenVals.add(val);
+            assertTrue(vals.contains(val));
+            seenCount++;
+        }
+        assertEquals(size, seenCount);
+    }
+
+    @Test
+    public void emptyToArray() {
+        testToArray(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallToArray() {
+        testToArray(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallToArray() {
+        testToArray(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularToArray() {
+        testToArray(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigToArray() {
+        testToArray(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigToArray() {
+        testToArray(SIZE_REALLY_BIG);
+    }
+
+    public void testContains(int size) {
+        System.out.println("Testing contains method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        if (size > 0) {
+            assertFalse(set.isEmpty());
+        } else {
+            assertTrue(set.isEmpty());
+        }
+        Collections.shuffle(vals);
+
+        for(ByteArray val: vals) {
+            assertTrue(set.contains(val));
+        }
+
+        List<ByteArray> absentVals = getRandomByteArrays(10);
+        for(ByteArray val: absentVals) {
+            if (vals.contains(val)) {
+                assertTrue(set.contains(val));
+            } else {
+                assertFalse(set.contains(val));
+            }
+        }
+    }
+
+    @Test
+    public void emptyContains() {
+        testContains(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallContains() {
+        testContains(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallContains() {
+        testContains(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularContains() {
+        testContains(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigContains() {
+        testContains(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigContains() {
+        testContains(SIZE_REALLY_BIG);
+    }
+
+    @Test
+    public void testNullEquals() {
```

Comment:
This test is not really testing anything. This will always return true because `null` will never equal `set = new ByteArraySet();` Also the `assertTrue` on the `set.add` is not getting us any extra test coverage than what we already have elsewhere here